src/compiler.rs

```diff
@@ -132,4 +152,36 @@ impl Compiler {
             file: out_file,
         }
     }
+
+    fn skip_compile(&self, out_file: String) -> CompileOutput {
+        CompileOutput {
+            status: ExitStatusExt::from_raw(0),
+            stdout: vec![],
+            stderr: vec![],
+            file: out_file,
+        }
+    }
+}
+
+fn is_source_modified(in_files: &[String], out_file: &str) -> bool {
```

Comment:
I think this function should be inside `impl Compiler`.

Or `get_paths_for` should be outside `impl Compiler`. Just to make things more consistent.

Reply:
It probably makes sense to be inside.